apps/src/applab/appStorage.js

```diff
@@ -266,14 +271,24 @@ AppStorage.onRecordEvent = function(tableName, onRecord, onError) {
     return;
   }
 
-  if (!recordListener.setListener(tableName, onRecord)) {
-    onError('You are already listening for events on table "' + tableName + '". ' +
-      'only one event handler can be registered per table.');
-  }
+  var table = getTable(Applab.channelId, tableName);
+  // CONSIDER: Do we need to make sure a client doesn't hear about updates that it triggered?
+
+  table.on('child_added', function(childSnapshot, prevChildKey) {
+    onRecord(childSnapshot.val(), 'create');
+  });
+
+  table.on('value', function(dataSnapshot) {
+    onRecord(dataSnapshot.val(), 'update');
+  });
```

Comment:
Note for later: This should probably be `table.on('child_changed', ...)` instead of `table.on('value', ...)`